jakarta-eclipse/src/org/jakarta/jdt/FilterDiagnosticsCollector.java

```diff
@@ -52,11 +57,40 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
 
 					
 					if (isWebFilterAnnotated && !isFilterImplemented) {
-						ISourceRange nameRange = JDTUtils.getNameRange(type);
-						Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
 						diagnostics.add(new Diagnostic(range, "Classes annotated with @WebFilter must implement the Filter interface."));
 					}
 					
+					/* URL pattern diagnostic check */
+					if(WebFilterAnnotation != null) {
+						IMemberValuePair[] memberValues = WebFilterAnnotation.getMemberValuePairs();
+						
+						boolean isUrlpatternSpecified = false;
+						boolean isServletNamesSpecified = false;
+						boolean isValueSpecified = false;
+						for (IMemberValuePair mv : memberValues) {
+							if(mv.getMemberName().equals(ServletConstants.URL_PATTERNS)) {
+								isUrlpatternSpecified = true;
+								continue;
+							}
+							if(mv.getMemberName().equals(ServletConstants.SERVLET_NAMES)) {
+								isServletNamesSpecified = true;
+								continue;
+							}
+							if(mv.getMemberName().equals(ServletConstants.VALUE)) {
+								isValueSpecified = true;
+							}
+							
+						}
+						
+						if (!isUrlpatternSpecified && !isValueSpecified && !isServletNamesSpecified) {
+							diagnostics.add(new Diagnostic(range, "The urlPatterns attribute, servletNames attribute or the value attribute of the annotation MUST be specified."));
```

Comment:
The range should be the range of the `@WebFilter` annotation, not the class name. 

The diagnostic message should be more specific to WebFilter, ie.
`The 'urlPatterns' attribute, 'servletNames' attribute or the 'value' attribute of the WebFilter annotation MUST be specified.`